front/src/app/scenes/transaction/transaction.component.ts

```diff
@@ -63,6 +70,46 @@ export class TransactionComponent implements OnInit, OnDestroy {
     this._layoutService.offLoading();
   }
 
+  processTransaction() {
+    if (!this.tx.parsedLogs.length) {
+      return;
+    }
+    forkJoin([
+      this._commonService.eventsAbi$,
+      this._commonService.getAddress(this.tx.to),
+    ]).subscribe(([events, address]: [ContractEventsAbi, Address]) => {
+      this.tx.addressData = address;
+      this.tx.processedLogs = this.tx.parsedLogs.map((log: TxLog) => {
+        const prettyLog: ProcessedLog = new ProcessedLog();
+        prettyLog.index = +log.logIndex;
+        prettyLog.contract_address = log.address;
+        prettyLog.removed = log.removed;
+        let abiItem: AbiItem;
+        const eventSignature = <string>log.topics[0];
+        const knownEvent = events[eventSignature];
+        if (knownEvent) {
+          const ercType = address.erc_types.find((item: string) => !!knownEvent[item]);
+          if (ercType) {
+            abiItem = knownEvent[ercType];
+          }
+        }
+        if (abiItem) {
+          log.topics.shift();
+          const decoded = this._walletService.w3.eth.abi.decodeLog(
+            abiItem.inputs,
+            log.data.replace('0x', ''),
+            <string[]>log.topics
+          );
+          const res: string[] = abiItem.inputs.map(input => `${input.name}: ${decoded[input.name]}`);
+          prettyLog.data = `${abiItem.name}(${res.join(', ')})`;
```

Comment:
It would be neat if values were links, but that might be messy, so it could wait for later.

Reply:
`address` might be easy to do now, since they are all the same. But some values require extra mapping, e.g. ERC20 Transfer uint256 is a value so a link wouldn't make sense, but ERC721 Transfer uint256 is a token ID and can be a special type of link.